crates/flux-desugar/src/resolver/refinement_resolver.rs

```diff
@@ -296,6 +296,17 @@ impl<V: ScopedVisitor> surface::visit::Visitor for ScopedVisitorWrapper<V> {
                 self.on_func(*func, expr.node_id);
             }
             surface::ExprKind::Dot(path, _) => self.on_path(path),
+            surface::ExprKind::Constructor(path, field_exprs, spread) => {
+                self.on_path(path);
+                // also need to deal with exprs in the constructor
+                // This is the case becasuse field exprs are different fields
+                for expr in field_exprs {
+                    self.visit_expr(&expr.expr);
+                }
```

Comment:
Yep this was because I made the mistake of calling `walk_expr` in `visit_expr`, when I should have been calling `vis.visit_expr` on field expressions.